cmd/notary-signer/main.go

```diff
@@ -73,28 +67,113 @@ func passphraseRetriever(keyName, alias string, createNew bool, attempts int) (p
 	return passphrase, false, nil
 }
 
-// parses and sets up the TLS for the signer http + grpc server
-func signerTLS(configuration *viper.Viper, printUsage bool) (*tls.Config, error) {
-	certFile := configuration.GetString("server.cert_file")
-	keyFile := configuration.GetString("server.key_file")
-	if certFile == "" || keyFile == "" {
-		if printUsage {
-			usage()
+// Reads the configuration file for storage setup, and sets up the cryptoservice
+// mapping
+func setUpCryptoservices(configuration *viper.Viper, allowedBackends []string) (
+	signer.CryptoServiceIndex, error) {
+
+	storeConfig, err := utils.ParseStorage(configuration, allowedBackends)
+	if err != nil {
+		return nil, err
+	}
+
+	var keyStore trustmanager.KeyStore
+	if storeConfig.Backend == utils.MemoryBackend {
+		keyStore = trustmanager.NewKeyMemoryStore(
+			passphrase.ConstantRetriever("memory-db-ignore"))
+	} else {
+		defaultAlias := configuration.GetString("storage.default_alias")
+		if defaultAlias == "" {
+			// backwards compatibility - support this environment variable
+			defaultAlias = configuration.GetString(defaultAliasEnv)
 		}
-		return nil, fmt.Errorf("Certificate and key are mandatory")
+
+		if defaultAlias == "" {
+			return nil, fmt.Errorf("must provide a default alias for the key DB")
+		}
+		logrus.Debug("Default Alias: ", defaultAlias)
+
+		dbSQL, err := sql.Open(storeConfig.Backend, storeConfig.Source)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open the %s database: %s, %v",
+				storeConfig.Backend, storeConfig.Source, err)
+		}
+		logrus.Debugf("Using %s DB: %s", storeConfig.Backend, storeConfig.Source)
+
+		keyStore, err := keydbstore.NewKeyDBStore(
```

Comment:
`:=` is going to create a keyStore variable scoped within this `else`. I think you just want `=`